### DISCUSSION TITLE
Help adding entities in a certain order

### DISCUSSION THREAD

**Discussion by Tonaion02** [bevyengine/bevy#15253] [Q&A]
I have to ask some questions about how Bevy works. The order in which i read entities from a query is defined by the order in which i add entities?

I am asking cause i need to retrieve entities in some order, and i know that is impossible to order entities for a query.

I thinked to write an exclusive system to add entities in the correct order, is it possible in some manners?

**Comment by SkiFire13** [comment]
The order in queries is deterministic and is in part influenced by the order you create entities, but is not equal to that order.

If you need to iterate over entities in a specific order consider storing a `Vec<Entity>` representing that order or using `QueryIter::sort_*` APIs

**Reply by Tonaion02** [reply]
But if in multiple runs i add the entities always in the same order, i read that always in the same order, right?

**Reply by SkiFire13** [reply]
Assuming no other factors (e.g. creating other entities, adding or removing components of other entities, etc etc), then you should see the same result if you do the same operations in the same order.

**Reply by Tonaion02** [reply]
Thx for the answer!
